### PR TITLE
Allow to use Origin API groups from legacy client

### PR DESCRIPTION
**Pull request by mfojtik**
This wrapper will allow to consume API groups using the Origin legacy client. It will allow the new resources to be consumed by internal controllers until we fix the clientset generators and switch the controller to use them. This should not be used for existing resources (!).

More specificaly this will be used by @jim-minter templatebroker PR where he is adding new resources that won't exists in the legacy Origin API, but only in the template.openshift.io group.

As noted in the comment, this is a temporary transition band-aid until we can switch to clientsets.

@deads2k @smarterclayton @sttts PTAL.

### CODE REVIEW

**Comment by mfojtik** [comment]
[test]

File: pkg/client/client.go [hunk 1]
```diff
@@ -325,7 +329,24 @@ func New(c *restclient.Config) (*Client, error) {
...
 
-	return &Client{client}, nil
+	return &Client{RESTClient: client, config: &config}, nil
+}
+
+// ForGroupVersion returns a copy of the client but without any openshift legacy
+// defaulting.
+// NOTE: This client should be only used in transition period when the legacy API is
+// locked for new types, but the clientsets are not provided.
+func (c *Client) ForGroupVersion(gv *unversioned.GroupVersion) *Client {
```
Comment (sttts) [inline pkg/client/client.go]: `ForGroupVersion(kapi.SchemeGroupVersion.GroupVersion())` will not be correct. But as this is only for transition we shouldn't care maybe.

File: pkg/client/client.go [hunk 2]
```diff
@@ -325,7 +329,24 @@ func New(c *restclient.Config) (*Client, error) {
...
+	newConfig := *c.config
+	newConfig.APIPath = "/apis"
+	newConfig.GroupVersion = gv
+
+	newClient, err := restclient.RESTClientFor(&newConfig)
+	if err != nil {
+		glog.Errorf("unable to get client for group %#v", gv)
```
Comment (sttts) [inline pkg/client/client.go]: print the error? Or even panic? At least, if we avoid adding error return types everywhere.

File: pkg/client/client.go (hunk 1 above)
Comment (mfojtik) [inline pkg/client/client.go]: @sttts i can guard against schema with empty group

File: pkg/client/client.go [hunk 3]
```diff
@@ -325,7 +328,26 @@ func New(c *restclient.Config) (*Client, error) {
 		return nil, err
 	}
 
-	return &Client{client}, nil
+	return &Client{RESTClient: client, config: &config}, nil
+}
+
+// ForGroupVersion returns a copy of the client but without any openshift legacy
```
Comment (deads2k) [inline pkg/client/client.go]: How is this client different than a client created by kube?  Couldn't we start collapsing onto theirs with moderate adjustments for GroupVersion?

Comment (mfojtik) [inline pkg/client/client.go]: @deads2k which kube client?

Comment (sttts) [inline pkg/client/client.go]: In fact, for everything new this is a very good idea 👍  There is no sense to produce legacy code if we can avoid it.

Comment (mfojtik) [inline pkg/client/client.go]: yeah, the dynamic client seems to be a good band-aid there, but having clientsets sounds better ;-)

**Comment by deads2k** [comment]
Opened https://github.com/openshift/origin/pull/13504 which generates current clientsets as an alternative

**Pull request closed without merging**